awsiotprotocol/loader_test.go

```diff
@@ -33,3 +34,22 @@ func TestByURL(t *testing.T) {
 		}
 	}
 }
+
+func TestByURLFailure(t *testing.T) {
+	testcase := []struct {
+		input                string
+		expectedErrorMessage string
+	}{
+		{input: "@@://@@@/@@.@@@", expectedErrorMessage: "parse @@://@@@/@@.@@@: "},
+		{input: "https://non-supported.protocol.com", expectedErrorMessage: "Protocol \"https\" is not supported"},
+	}
+	for _, v := range testcase {
+		_, err := ByURL(v.input)
+		if err == nil {
+			t.Errorf("awsiotprotocol.ByURL should fail with invalid input.\ninput: %#v", v.input)
+		}
+		if !strings.Contains(err.Error(), v.expectedErrorMessage) {
```

Comment:
```suggestion
		else if !strings.Contains(err.Error(), v.expectedErrorMessage) {
```
`err.Error()` causes segmentation fault if `err == nil`

Reply:
Ah, now I understand your previous comment and what go faq says. These cases should never crash during the test run (even when test fails). Thank you for the suggestion!